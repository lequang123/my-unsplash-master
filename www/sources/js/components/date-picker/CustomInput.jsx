import React, { Fragment } from 'react';
import { Form } from 'react-bootstrap';

export default class CustomInput extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        const { disabled, onClick, value } = this.props;

        return (
            <Fragment>
                <Form.Control as='input' readOnly disabled={disabled} onClick={onClick} value={value} />
                <i className='date-picker-icon fa fa-calendar' onClick={onClick} />
            </Fragment>
        );
    }
}