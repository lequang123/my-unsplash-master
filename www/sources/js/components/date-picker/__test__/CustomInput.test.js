import React from 'react';
import CustomInput from '../CustomInput'
import { Form } from 'react-bootstrap';
import { shallow } from 'enzyme';

describe('CustomInput', () => {
    it('should have correct snapshots', () => {
        // Action
        const customInput = shallow(<CustomInput />);

        // Assert
        expect(customInput).toMatchSnapshot();
    });

    it('render should return correct component', () => {
        // Arrange
        const props = {
            disabled: true,
            onClick: jest.fn(),
            value: 'value'
        };

        // Action
        const customInput = shallow(<CustomInput {...props} />);

        // Assert
        expect(customInput.find(Form.Control).props().as).toBe('input');
        expect(customInput.find(Form.Control).props().readOnly).toBeTrue;
        expect(customInput.find(Form.Control).props().disabled).toEqual(props.disabled);
        expect(customInput.find(Form.Control).props().onClick).toEqual(props.onClick);
        expect(customInput.find(Form.Control).props().value).toEqual(props.value);
        expect(customInput.find('i').props().className).toBe('date-picker-icon fa fa-calendar');
        expect(customInput.find('i').props().onClick).toEqual(props.onClick);
    });
});