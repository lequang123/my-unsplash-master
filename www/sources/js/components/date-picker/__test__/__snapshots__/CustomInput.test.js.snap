// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput should have correct snapshots 1`] = `ShallowWrapper {}`;
